hyperSpec/R/bind.R

```diff
@@ -169,8 +204,63 @@ rbind.hyperSpec <- function(...) bind("r", ...)
     expect_equal(nrow(rbind(flu, flu)), 2 * nrow(flu))
     expect_error(rbind(flu, flu[, , min ~ min + 3i]))
   })
+
+  test_that("rbind.hyperSpec() uses bind('r', ...) correctly", {
+    expect_equal(bind("r", flu, flu), rbind(flu, flu))
+  })
+
+  test_that("rbind.hyperSpec() works", {
+
+    # One dataset
+    expect_equal(rbind(flu), flu)
+
+    # Two datasets
+    expect_equal(rbind(flu[1:3, , ], flu[4:6, , ]), flu)
+    expect_equal(
+      rbind(flu[c(1, 3, 6), , ], flu[c(2, 4, 5), , ]),
+      flu[c(1, 3, 6, 2, 4, 5), , ]
+    )
+
+    # Three datasets
+    expect_equal(
+      rbind(flu[1, , ], flu[6, , ], flu[5, , ]),
+      flu[c(1, 6, 5), , ]
+    )
+  })
+}
+
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+.test(cbind.hyperSpec) <- function() {
+  context("cbind.hyperSpec")
+
+  test_that("cbind.hyperSpec() uses bind('c', ...) correctly", {
+    expect_equal(bind("c", flu, flu), cbind(flu, flu))
+  })
+
+  test_that("cbind.hyperSpec() works", {
+
+    # One dataset
+    expect_equal(cbind(flu), flu)
+
+    # Two datasets
+    expect_equal(
+      cbind(flu[, , 1:10, wl.index = TRUE], flu[, , 21:30, wl.index = TRUE]),
+      flu[, , c(1:10, 21:30), wl.index = TRUE]
+    )
+
+    # Three datasets
+    expect_equal(
+      cbind(
+        flu[, , 1:10, wl.index = TRUE],
+        flu[, , 21:30, wl.index = TRUE],
+        flu[, , 51:80, wl.index = TRUE]
+      ),
+      flu[, , c(1:10, 21:30, 51:80), wl.index = TRUE]
+    )
+  })
```

Comment:
@cbeleites, see if this addresses in https://github.com/cbeleites/hyperSpec/pull/225#discussion_r462748907

> We need tests whether the results for more than one object to be r/cbound are correct.